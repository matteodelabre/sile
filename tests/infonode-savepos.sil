\begin[papersize=a8]{document}
  \use[module=packages/infonode]

  \nofolios
  \neverindent

  \begin{script}
    SILE.documentState.documentClass:registerHook("endpage", function ()
      -- Draw a thin red line linking the info nodes
      local command = {"0.5 w", "1 0 0 RG"}

      for index, node in ipairs(SILE.scratch.info.thispage.line) do
        local x = SU.cast("number", node.x)
        local y = SU.cast("number", node.y)
        table.insert(command, string.format("%.4f %.4f", x, y))

        if index == 1 then
          table.insert(command, "m")
        else
          table.insert(command, "l")
        end
      end

      table.insert(command, "S")
      SILE.outputter:drawSVG(table.concat(command, " "), 0, 0, 0, 0, 1)
    end)
  \end{script}

  first line\info[category="line", savepos=true]

  \info[category="line", savepos=true]second line

  third line\info[category="line", savepos=true]

  \info[category="line", savepos=true]fourth line

  fifth line\info[category="line", savepos=true]
\end{document}
