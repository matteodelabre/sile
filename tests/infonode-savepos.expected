Set paper size 	147.4015764	209.7637818
Begin page
Mx 	7.3701
My 	34.4882
Set font 	Gentium Plus;10;400;;normal;;LTR
T	737 85 86 87 w=17.0850	(first)
Mx 	27.1083
T	79 76 81 72 w=15.5566	(line)
Mx 	7.3701
My 	46.4882
T	86 72 70 82 81 71 w=28.5986	(second)
Mx 	38.6207
T	79 76 81 72 w=15.5566	(line)
Mx 	7.3701
My 	58.4882
T	87 75 76 85 71 w=20.8301	(third)
Mx 	30.8530
T	79 76 81 72 w=15.5566	(line)
Mx 	7.3701
My 	70.4882
T	73 82 88 85 87 75 w=26.3574	(fourth)
Mx 	36.3798
T	79 76 81 72 w=15.5566	(line)
Mx 	7.3701
My 	82.4882
T	737 73 87 75 w=17.8906	(fifth)
Mx 	27.9139
T	79 76 81 72 w=15.5566	(line)
Draw SVG	0.0000	0.0000	0.0000	1.0000	0.5 w 1 0 0 RG 42.6650 34.4882 m 7.3701 46.4882 l 46.4097 58.4882 l 7.3701 70.4882 l 43.4706 82.4882 l S
End page
Finish
